pkg/controller/kataconfig/openshift_reconciler.go

```diff
@@ -500,7 +504,10 @@ func (r *ReconcileKataConfigOpenShift) processKataConfigInstallRequest() (reconc
 		if err != nil && errors.IsNotFound(err) {
 			r.kataLogger.Info("Creating a new installation Daemonset", "ds.Namespace", ds.Namespace, "ds.Name", ds.Name)
 			err = r.client.Create(context.TODO(), ds)
-			if err != nil {
+			/* Retry when there was a conflict */
+			if err != nil && apimachineryErrors.IsConflict(err) {
+				return reconcile.Result{}, nil
```

Comment:
If you get a conflict, shouldn't you reschedule another reconciliation? Since it's a conflict, your object has not been modified. I think you will have to request the reconciliation.

Reply:
the comment you added `/* Retry when there was a conflict */` means you intend to retry, but if you return `reconcile.Result{}, nil` the controller will not reschedule another reconciliation.

Reply:
You're right. It should be reconcile.Result{Requeue: true,...} or err has to be != nil. Thanks!